app/models/utility/cell_count_spot_checking.rb

```diff
@@ -0,0 +1,144 @@
+# frozen_string_literal: true
+
+module Utility
+  # This class is used to select wells from a plate for spot checking.
+  # NB. It assumes that the barcodes of the ancestor tubes are stored in the
+  # sample metadata supplier_name field.
+  class CellCountSpotChecking
+    attr_reader :plate, :ancestor_tubes
+
+    # Initializes the object with a plate. It is use to pick wells from the
+    # plate for spot checking.
+    #
+    # @param plate [Plate] the plate associated with this instance
+    def initialize(plate)
+      @plate = plate
+    end
+
+    # Returns the number of ancestor tubes associated with the plate. It finds
+    # the number of unique supplier names in the sample metadata of the first
+    # sample in each well.
+    #
+    # @return [Integer] the number of ancestor tubes
+    def ancestor_tubes_size
+      @ancestor_tubes_size ||= filtered_wells.map { |well| ancestor_barcode(well) }.uniq.size
+    end
+
+    # Selects at most the specified number of wells from the plate. If the
+    # number of ancestor tubes is less than the count, or if the count is
+    # not specified, it selects one well per ancestor tube.
+    #
+    # @param count [Integer] the number of wells to select
+    # @return [Array<Well>] an array of selected Well objects
+    def select_wells(count = nil)
+      count ||= ancestor_tubes_size
```

Comment:
I think that `||=` here has no benefit over `=`, because you're not saving it outside the scope of this method?

Normally it would be saved into an instance variable, and the method would be called more than once.

Reply:
It is used twice within the same scope. The first use is setting a default for count (all tubes if not given), the second is for honouring an explicit count if given by taking the min of count or ancestor_tubes_size. So, there is a performance benefit in caching it. That is why it is done. One could create another variable and store it separately as well but this shortens the code.

Reply:
I might have misunderstood it but still it is necessary. It is setting a default for the count if not given.